### PR TITLE
Unified internal and external Kotlin rules

### PR DESCRIPTION
**Pull request by piotrswigon**

### CODE REVIEW

File: third_party/java/arcs/build_defs/internal/kotlin.bzl [hunk 1]
```diff
@@ -24,19 +24,21 @@ def arcs_kt_binary(name, srcs = [], deps = [], visibility = None):
     """Performs final compilation of wasm and bundling if necessary."""
     libname = name + "_lib"
```
Comment (csilvestrini) [inline third_party/java/arcs/build_defs/internal/kotlin.bzl]: Minor nit: move this def inside the if-statement below

File: third_party/java/arcs/build_defs/internal/kotlin.bzl [hunk 2]
```diff
@@ -24,19 +24,21 @@ def arcs_kt_binary(name, srcs = [], deps = [], visibility = None):
...
+    if srcs:
+        # Declare a library because g3 kt_native_binary doesn't take srcs
+        kt_native_library(
+            name = libname,
+            srcs = srcs,
+            deps = _ARCS_KOTLIN_LIBS + deps,
+            visibility = visibility
+        )
+
+        deps = [":%s" % libname] + deps
```
Comment (csilvestrini) [inline third_party/java/arcs/build_defs/internal/kotlin.bzl]: I think you can just set `deps  = [":" + libname]` here. The binary depends only on the library; the library depends on everything else.

Comment (cromwellian) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl]: I think I tried that before and there was some strict_deps issue, im ok with removing it, but test internally first.

Comment (piotrswigon) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl]: Done

File: third_party/java/arcs/build_defs/internal/kotlin.bzl (hunk 1 above)
Comment (piotrswigon) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl]: Done

File: third_party/java/arcs/build_defs/internal/kotlin.bzl (hunk 2 above)
Comment (piotrswigon) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl]: Done

File: third_party/java/arcs/build_defs/internal/kotlin.bzl (hunk 1 above)
Comment (piotrswigon) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl]: Done

**Pull request merged**